### PR TITLE
feat: Allow setting scriptName $0

### LINKED ISSUE #1010: Question: Filename appearing next to command in help menu
Just a quick question. I am not sure if this is a NodeJS issue or what, but when I run the example yargs code, I get the following in my terminal:

```
example.js <cmd> [args]

Commands:
  example.js hello [name]  welcome ter yargs!

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
```

In the example, the filename is not next to the command `hello` under the `Commands` section. Just wondering how to get the filename out of that section so that it looks like the one from the example:

```
  Commands:
    hello [name]  welcome ter yargs!

```

### PR DESCRIPTION
**Pull request by laggingreflex**
Fixes #1010 #1048 #1065

### CODE REVIEW

**Comment by axe312ger** [comment]
bump!

**Comment by axe312ger** [comment]
@bcoe this PR even has documentation 🙈

File: yargs.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ function Yargs (processArgs, cwd, parentRequire) {
 
   if (!cwd) cwd = process.cwd()
 
+  self.scriptName = function scriptName ($0) {
```
Comment (evocateur) [inline yargs.js]: No reason we can't use an intelligible parameter name here

Comment (laggingreflex) [reply yargs.js]: Saw it being used throughout (like [#L899](../../blob/master/yargs.js#L899)) so just went with it, but yeah I agree, shall change it to `scriptName`

**Comment by bcoe** [comment]
👋 hey, still haven't had a chance to sit down and grind through any OSS, we're having an all hands all week at npm, Inc. _But_, will hopefully land some tickets this weekend.

**Comment by bcoe** [comment]
@laggingreflex @axe312ger love it 👍

**Comment by bcoe** [comment]
@laggingreflex `scriptName` is now available on `npm i yargs@next` mind trying it out and letting me know if it works as expected.

**Pull request merged**